### DISCUSSION TITLE
Is it posible to use `suspense` library in the next js app router

### DISCUSSION THREAD

**Discussion by jeetiss** [bvaughn/suspense#36] [Q&A] [answered]
Next js replace `react@18.2` with the experimental build so it should be possible to use `suspense` with app router. am I right? 

<img width="509" alt="Снимок экрана 2023-06-22 в 12 58 52" src="https://github.com/bvaughn/suspense/assets/6726016/3021a908-a7da-425d-87a1-971b91c72beb">

**Comment by bvaughn** [comment, marked as answer]
This package requires the _experimental_ release channel.

From the [React announcement post](https://react.dev/blog/2023/05/03/react-canaries) about release channels:
> * Unlike the [Experimental](https://react.dev/community/versioning-policy#experimental-channel) channel, React Canaries only include features that we reasonably believe to be ready for adoption. We encourage frameworks to consider bundling pinned Canary React releases.

I believe the canary channel does have `unstable_useCacheRefresh ` but it does not yet have the `unstable_getCacheForType` which this package relies on for mutation support.

**Reply by jeetiss** [reply]
I found in the `d.ts` file that `serverActions: true` enables `experimental` channel and i managed to use `suspense` library with the next 🙌🏻

```js
/**
 * Enable `react@experimental` channel for the `app` directory.
 */
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    serverActions: true,
  },
};
```

**Reply by bvaughn** [reply]
Nice find! I’m glad you got it sorted 😌

I would love to know what you’re building!

**Reply by bvaughn** [reply]
Stumbled across this problem myself today and referred back to this thread 🥳
